src/main/java/net/fabricmc/loader/impl/metadata/V1ModMetadataParser.java

```diff
@@ -398,6 +402,25 @@ private static void readMixinConfigs(List<ParseWarning> warnings, JsonReader rea
 					case "environment":
 						environment = V1ModMetadataParser.readEnvironment(reader);
 						break;
+					case "depends":
+						if (reader.peek() != JsonToken.BEGIN_ARRAY) {
+							throw new ParseMetadataException("Value of \"depends\" must be an array", reader);
+						}
+
+						reader.beginArray();
+						depends = new HashSet<>();
+
+						while (reader.hasNext()) {
+							if (reader.peek() == JsonToken.STRING) {
+								depends.add(reader.nextString());
+							} else {
+								warnings.add(new ParseWarning(reader.getLineNumber(), reader.getColumn(), "Invalid mixin config dependency type"));
```

Comment:
This can probably be an exception as well if you fail when not reading an array.

Reply:
It might be better to replace the exception with a warning too and skip the whole element if it's not an array. People can get a proper dependency error from using a Loader version that is too old with a warning, rather than a crash trying to read the `fabric.mod.json`. Of course I strongly doubt anyone is using the key for anything right now, but it keeps our options open in the future.

Reply:
Changed the exception to a warning